pkg/apiserver/storage/testing/store_tests.go

```diff
@@ -290,6 +290,11 @@ func RunTestConditionalDelete(ctx context.Context, t *testing.T, store storage.I
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			if tt.name == "UID match" {
+				// The UID changes on CREATE
+				tt.precondition = storage.NewUIDPreconditions(string(storedObj.UID))
+			}
+
```

Comment:
why are the changes in this file needed?

Reply:
etcd and file store let the external request set the UID

The new backend creates the UID internally on CREATE.  These tests assume assume the value was created with UID=A and then tests that.  The change updates the test to match the UID that was created

Reply:
> The new backend creates the UID internally on CREATE. These tests assume assume the value was created with UID=A and then tests that.

it seems like that it might not just be tests that expect that behavior based on how the [generic store creates metadata](https://github.com/kubernetes/kubernetes/blob/b785c47f6da0dd5a9f63d11ac0a4d46722124c65/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L465) and comments [like this](https://github.com/kubernetes/kubernetes/blob/b785c47f6da0dd5a9f63d11ac0a4d46722124c65/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L485-L486).